resource_booking/models/resource_booking_type.py

```diff
@@ -0,0 +1,149 @@
+        required=True,
+        default=24,
+        help=(
+            "When this deadline has been exceeded, if a booking was not yet "
+            "confirmed, it will be canceled automatically. Also, only booking "
+            "managers will be able to unschedule or reschedule them. "
+            "The value is expressed in hours."
+        ),
+    )
+    name = fields.Char(index=True, translate=True, required=True)
+    booking_ids = fields.One2many(
+        comodel_name="resource.booking",
+        inverse_name="type_id",
+        string="Bookings",
+        help="Bookings available for this type",
+    )
+    resource_calendar_id = fields.Many2one(
+        comodel_name="resource.calendar",
+        default=lambda self: self._default_resource_calendar(),
+        index=True,
+        required=True,
+        ondelete="restrict",
+        string="Availability Calendar",
+        help="Restrict bookings to this schedule.",
+    )
+    requester_advice = fields.Text(
+        translate=True,
+        help=(
+            "Text that will appear by default in portal invitation emails "
+            "and in calendar views for scheduling."
+        ),
+    )
+
+    @api.model
+    def _default_resource_calendar(self):
+        return self.env.company.resource_calendar_id
+
+    @api.depends("booking_ids")
+    def _compute_booking_count(self):
+        data = self.env["resource.booking"].read_group(
+            [("type_id", "in", self.ids)], ["type_id"], ["type_id"]
+        )
+        mapping = {x["type_id"][0]: x["type_id_count"] for x in data}
+        for one in self:
+            one.booking_count = mapping.get(one.id, 0)
+
+    @api.constrains("booking_ids", "resource_calendar_id", "combination_rel_ids")
+    def _check_bookings_scheduling(self):
+        """Scheduled bookings must have no conflicts."""
+        bookings = self.mapped("booking_ids")
+        return bookings._check_scheduling()
+
+    def _get_combinations_priorized(self):
+        """Gets all combinations sorted by the chosen assignment method."""
+        if not self.combination_assignment:
+            return self.combination_rel_ids.mapped("combination_id")
+        keys = {"sorted": "sequence", "random": lambda *a: random()}
+        rels = self.combination_rel_ids.sorted(keys[self.combination_assignment])
+        combinations = rels.mapped("combination_id")
+        return combinations
+
+    def action_open_bookings(self):
+        FloatTimeParser = self.env["ir.qweb.field.float_time"]
+        return {
+            "context": dict(
+                self.env.context,
+                default_alarm_ids=[(6, 0, self.alarm_ids.ids)],
+                default_description=self.requester_advice,
+                default_duration=self.duration,
+                default_type_id=self.id,
+                # Context used by web_calendar_slot_duration module
+                calendar_slot_duration=FloatTimeParser.value_to_html(
+                    self.duration, False
```

Comment:
```suggestion
                    self.slot_duration, False
```
You added a slot_duration field intentionally to separe the duration of the booking and the slot in calendar, so this change is consequent. But anyway this change is uselles because without the dependency of `web_calendar_slot_duration` this doesn't work. 
If you look at my PR: https://github.com/OCA/web/pull/2646 , you must notice that the slot_duration that expects is in this format: (00:00:00), but this FloatTimeParser function returns it in this one: (00:00). So you will have to make a conversion from the Float to this first format.

Reply:
Is (00:00:00) new for v16?
If yes, can you help me with this conversion?
If no, can you show me where the conversion is done for self.duration?